File: webhook.php
<?php
$VERIFY_TOKEN = "agrachat_test";

// Handle verification (GET request)
if ($_SERVER['REQUEST_METHOD'] === 'GET') {
    if ($_GET['hub_verify_token'] === $VERIFY_TOKEN) {
        echo $_GET['hub_challenge'];
        http_response_code(200);
        exit;
    } else {
        http_response_code(403); // Forbidden
        echo "Forbidden";
        exit;
    }
}

// Handle incoming webhook events (POST request)
if ($_SERVER['REQUEST_METHOD'] === 'POST') {
    // Get the raw POST data
    $input = file_get_contents('php://input');
    $data = json_decode($input, true);

    // Log the event for debugging
    file_put_contents("webhook_log.txt", print_r($data, true), FILE_APPEND);

    // Respond to Facebook
    echo "OK";
    http_response_code(200);
    exit;
}
?>

File: webhook.py
from flask import Flask, request, jsonify

# Initialize Flask app
app = Flask(__name__)

# Your custom verification token
VERIFY_TOKEN = "agrachat_test"

@app.route('/webhook', methods=['GET'])
def verify():
    """
    Verifies the webhook with Facebook during the setup.
    Facebook sends a GET request with hub.verify_token and hub.challenge.
    If the token matches, respond with hub.challenge.
    """
    token = request.args.get('hub.verify_token')
    challenge = request.args.get('hub.challenge')
    if token == VERIFY_TOKEN:
        return challenge, 200
    return "Forbidden", 403

@app.route('/webhook', methods=['POST'])
def handle_webhook():
    """
    Handles incoming webhook events from Facebook.
    Logs the data for debugging or further processing.
    """
    data = request.json
    print("Received Webhook Event:", data)

    # Example: process comment or message (add logic here as needed)
    # if 'entry' in data:
    #     for entry in data['entry']:
    #         # Handle specific event types here
    #         pass

    return "OK", 200

if __name__ == "__main__":
    # Run the Flask app on port 5000
    app.run(host='0.0.0.0', port=5000)
